orm/src/main/java/org/hibernate/search/FullTextQuery.java

```diff
@@ -29,7 +28,7 @@
  * @author Hardy Ferentschik
  * @author Emmanuel Bernard
  */
-public interface FullTextQuery<R> extends Query<R>, ProjectionConstants, QueryImplementor<R> {
```

Comment:
Interesting, I'm going to try that as it looks promising. FYI this wasn't possible in earlier versions of ORM 5.2.x, but those SPIs evolved a bit also based on our earlier attempts to migrate Search..